src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -226,8 +228,8 @@ public TopicPartitionChannel(
       KafkaRecordErrorReporter kafkaRecordErrorReporter,
       SinkTaskContext sinkTaskContext,
       SnowflakeConnectionService conn) {
-    this.streamingIngestClient = Preconditions.checkNotNull(streamingIngestClient);
-    Preconditions.checkState(!streamingIngestClient.isClosed());
+    // TODO: there is no guarantee that the ingestsdkprovider has already been initialized. could lead to task creation failures
```

Comment:
The following comment states that tasks may be started before/during connector start. Seems like we haven't ran into any issues with that yet, but it is something to think about. Moving the client creation from Task.start to Connector.start may exacerbate this issue. 

We already have a boolean in the Connector to indicate when everything has been initialized, perhaps we want to expand the scope of this boolean to here as well. I don't like it, but I'm not sure there are many other good options

https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java#L65